docs/pyasdf/context_managers.rst

```diff
@@ -0,0 +1,43 @@
+.. _context-managers:
+
+Context managers
+================
+
+`~pyasdf.AsdfFile` objects can be used as context managers to be used
+with ``with`` statements to help ensure that the underlying files are
+automatically closed.  There is no single pattern to recommend when
+using `~pyasdf.AsdfFile` objects as context managers, but it will
+instead depend on the context (no pun intended) in which they are
+used.
+
+For example, basic writing and reading of a file::
+
+  >>> import pyasdf
+  >>> tree = {'key': 'value'}
+  >>> with pyasdf.AsdfFile(tree).write_to('test.asdf'):
+  ...     pass
+
+  >>> with pyasdf.AsdfFile.read('test.asdf') as ff:
+  ...     print(ff.tree['key'])
+
+The reason that `~pyasdf.AsdfFile.write_to` and
+`~pyasdf.AsdfFile.read` functions do not automatically close the file
+(unless used with ``with`` statements), is to allow for updating the
+file in-place later on.  For example::
+
+  >>> with pyasdf.AsdfFile.read('test.asdf', 'rw') as ff:
+  ...     ff.tree['key'] = 'new value'
+  ...     ff.update()
```

Comment:
Part of me wonders if this example wouldn't make more sense as ``open`` instead of ``read`` (kind of like ``astropy.io.fits.open). That is:

```
>>> with pyasdf.AsdfFile.open('test.asdf', 'rw') as ff:
  ...     ff.tree['key'] = 'new value'
  ...     ff.update()
```

because I find it confusing to read something with read/write mode.